app/src/main/java/dev/nadeldrucker/trafficswipe/fragments/StartFragment.java

```diff
@@ -60,25 +64,35 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
 
             @Override
             public void afterTextChanged(Editable s) {
+                final String input = etSearch.getText().toString().trim();
 
-                previewString(etSearch.getText().toString(), view.findViewById(R.id.tvChar1), view.findViewById(R.id.tvChar2), view.findViewById(R.id.tvChar3));
-                if (etSearch.getText().toString().length() >= 3) {
-                    Log.d(TAG, "edit text is done! " + etSearch.getText().toString());
+
+                previewString(input, view.findViewById(R.id.tvChar1), view.findViewById(R.id.tvChar2), view.findViewById(R.id.tvChar3));
+                if (input.length() >= 3) {
+                    searchViewModel.getQuery().postValue(input);
+                    lastQuery = input.toUpperCase();
+
+                    Log.d(TAG, "edit text is done! " + s);
                     setSoftKeyboardState(false, etSearch);
-                    showDepartureTable(etSearch.getText().toString());
+
                     etSearch.clearFocus();
                     etSearch.getText().clear();
+
+
+
                 }
             }
         });
 
+        searchViewModel.getAbbreviations().observe(Objects.requireNonNull(getActivity()), this::handleResult);
```

Comment:
This gets called **everytime** the abbreviations change, e.g. when you start the search. Better don't mix up view models. 
Better put the check inside the DeparturesViewModel for seperation of concerns.
Application logic should never live inside the Views, only inside ViewModels.